File: material/quad/quad0.c
#include <stdio.h>
#include <math.h>

int main(void)
{
    float a = 3.0f;
    float b = -2.0f;
    float c = -3.0f;
    float r1, r2;
    float det = b * b - 4.0f * a * c;
    int num_roots;
    if(det < 0.0f)
        num_roots = 0;
    r1 = (-b + sqrt(det)) / (2.0f * a);
    if(det == 0.0f)
        num_roots = 1;
    r2 = (-b - sqrt(det)) / (2.0f * a);
    num_roots = 2;
    printf("Number of roots: %d\n", num_roots);
    if(num_roots == 1)
        printf("Root: %.2f\n", r1);
    else if(num_roots == 2)
        printf("Root 1: %.2f - Root 2: %.2f\n", r1, r2);
}

File: material/quad/quad0.py
import math

a = 3.0
b = -2.0
c = -3.0
det = b * b - 4.0 * a * c
if det < 0:
    num_roots = 0
r1 = (-b + math.sqrt(det)) / (2.0 * a)
if det == 0:
    num_roots = 1
else:
    num_roots = 2
r2 = (-b - math.sqrt(det)) / (2.0 * a)
print '%d roots' % num_roots
if num_roots == 1:
    print 'Root: %.2f' % r1
elif num_roots == 2:
    print "Root 1: %.2f - Root 2: %.2f\n" % (r1, r2)
